packages/typespec-ts/src/modular/helpers/typeHelpers.ts

```diff
@@ -1,155 +1,226 @@
 import { EnumValue, Type } from "../modularCodeModel.js";
 
+/**
+ * Represents metadata for a given Type to generate the TypeScript equivalent.
+ */
 export interface TypeMetadata {
   name: string;
   originModule?: string;
   isRelative?: boolean;
-  modifier?: "Array";
-}
-
-function getNullableType(name: string, type: Type): string {
-  if (type.nullable) {
-    return `(${name} | null)`;
-  }
-
-  return name;
+  nullable?: boolean;
+  modifier?: "Array" | "NullableArray";
 }
 
+/**
+ * Returns a string representation for an anonymous enum using its values.
+ */
 function getAnonymousEnumName(values: EnumValue[]): string {
   return values
     .map((v) => (typeof v.value === "string" ? `"${v.value}"` : `${v.value}`))
     .join(" | ");
 }
 
+// Mapping of simple types to their TypeScript equivalents.
+const simpleTypeMap: Record<string, TypeMetadata> = {
+  Key: {
+    name: "KeyCredential",
+    originModule: "@azure/core-auth",
+    isRelative: false
+  },
+  OAuth2: {
+    name: "TokenCredential",
+    originModule: "@azure/core-auth",
+    isRelative: false
+  },
+  boolean: { name: "boolean" },
+  datetime: { name: "Date" },
+  float: { name: "number" },
+  integer: { name: "number" },
+  "byte-array": { name: "Uint8Array" },
+  string: { name: "string" },
+  any: { name: "Record<string, any>" },
+  unknown: { name: "unknown" }
+};
+
+/**
+ * Maps a given Type to its TypeScript representation metadata.
+ */
 export function getType(type: Type, format?: string): TypeMetadata {
-  switch (type.type) {
-    case "Key":
-      return {
-        name: "KeyCredential",
-        originModule: "@azure/core-auth",
-        isRelative: false
-      };
-    case "OAuth2":
-      return {
-        name: "TokenCredential",
-        originModule: "@azure/core-auth",
-        isRelative: false
-      };
-    case "boolean":
-      return { name: getNullableType(type.type, type) };
-    case "constant": {
-      let typeName: string = type.value?.toString() ?? "undefined";
-      if (type.valueType?.type === "string") {
-        typeName = type.value ? `"${type.value}"` : "undefined";
-      }
-      return { name: getNullableType(typeName, type) };
+  // Handle simple type conversions
+  const simpleType = simpleTypeMap[type.type];
```

Comment:
should we return nullable in simpleType so that we don't add the below code ?

Reply:
yup, updated